californium-core/src/main/java/org/eclipse/californium/core/network/BaseMatcher.java

```diff
@@ -158,7 +163,8 @@ protected final Exchange matchNotifyResponse(final Response response, final Corr
 		final Exchange.KeyToken idByToken = Exchange.KeyToken.fromInboundMessage(response);
 		Exchange exchange = null;
 
-		final Observation obs = observationStore.get(response.getToken());
+		final EndpointIdentifier sourceEndpoint = response.getSourceEndpoint();
```

Comment:
The `IdentifierFactory` is responsible to translate Identifier in context and context in Identifier.
(translation is centralized in this class)

I have in mind that context could be a connector concept and Identifier a CoAP stack one.